livestream-compose-sample/src/main/java/io/getstream/livestream/compose/LiveStreamScreen.kt

```diff
@@ -0,0 +1,142 @@
+package io.getstream.livestream.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.material.Icon
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Keyboard
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.compose.handlers.SystemBackPressedHandler
+import io.getstream.chat.android.compose.state.messages.Thread
+import io.getstream.chat.android.compose.ui.messages.composer.MessageComposer
+import io.getstream.chat.android.compose.ui.messages.composer.components.MessageInput
+import io.getstream.chat.android.compose.ui.messages.header.MessageListHeader
+import io.getstream.chat.android.compose.ui.messages.list.MessageList
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.viewmodel.messages.MessageComposerViewModel
+import io.getstream.chat.android.compose.viewmodel.messages.MessageListViewModel
+
+
+@Composable
+fun LiveStreamScreen(
+    composerViewModel: MessageComposerViewModel,
+    listViewModel: MessageListViewModel,
+    onBackPressed: () -> Unit = {}
+) {
+    val currentState = listViewModel.currentMessagesState
+    val messageMode = listViewModel.messageMode
+    val isNetworkAvailable by listViewModel.isOnline.collectAsState()
+    val user by listViewModel.user.collectAsState()
+    LaunchedEffect(Unit) {
+        listViewModel.start()
+    }
+
+    val backAction = {
+        val isInThread = listViewModel.isInThread
+        val isShowingOverlay = listViewModel.isShowingOverlay
+
+        when {
+            isShowingOverlay -> listViewModel.selectMessage(null)
+            isInThread -> {
+                listViewModel.leaveThread()
+                composerViewModel.leaveThread()
+            }
+            else -> onBackPressed()
+        }
+    }
+
+    SystemBackPressedHandler(isEnabled = true, onBackPressed = backAction)
+
+    ChatTheme(isInDarkMode = true) {
```

Comment:
As per the comment before, we shouldn't do this here, we should use it as a top-level component.